### PR TITLE
EDSC-3598: Convert class components to functional components (GranuleImage)

### PR DESCRIPTION
**Pull request by eudoroolivares2016**
# Overview

### What is the feature?

Convert class components to functional components (GranuleImage)

### What is the Solution?

Changed the class component to use functional components and changed the accompanying test to use
react-testing-library instead of enzyme

### What areas of the application does this impact?

Granule images

# Testing

### Reproduction steps
Select a collection that has images
select a granule and see if the granule image component comes up
ensure that the open and close functionality is working
ensure that loading dots appear if you try to switch between granules and ensure that these
load correctly

### Attachments

Please include relevant screenshots or files that would be helpful in reviewing and verifying this change.

# Checklist

- [x] I have added automated tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

### CODE REVIEW

File: static/src/js/components/GranuleImage/GranuleImage.js [hunk 1]
```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
...
+    setIsErrored(false)
+  })
+
+  const handleImageErrored = (() => {
+    setIsLoaded(false)
+    setIsLoading(false)
+    setIsErrored(true)
+  })
+
+  const loadImage = () => {
```
Comment (macrouch) [inline static/src/js/components/GranuleImage/GranuleImage.js:51]: `loadImage` is a confusing name, `handleImageLoading` might make more sense, and follows the other function names above. 

But it looks like the only time this function is called is within the useEffect, can you just call the 3 state updates within the useEffect?

File: static/src/js/components/GranuleImage/__tests__/GranuleImage.test.js [hunk 2]
```diff
@@ -1,70 +1,68 @@
...
-  }
-
-  const enzymeWrapper = shallow(<GranuleImage {...props} />)
-
-  return {
-    enzymeWrapper,
-    props
-  }
+const setup = (props) => {
+  act(() => {
```
Comment (macrouch) [inline static/src/js/components/GranuleImage/__tests__/GranuleImage.test.js:12]: Is this `act` necessary?

File: static/src/js/components/GranuleImage/__tests__/GranuleImage.test.js [hunk 3]
```diff
@@ -1,70 +1,68 @@
...
     test('renders itself correctly', () => {
-      const { enzymeWrapper } = setup({
+      const props = {
         imageSrc: '/some/image/src'
-      })
-
-      expect(enzymeWrapper.type()).toBe('div')
+      }
+      setup(props)
+      expect(screen.queryByRole('button')).toBeTruthy()
```
Comment (macrouch) [inline static/src/js/components/GranuleImage/__tests__/GranuleImage.test.js:36]: We generally prefer to use `screen.getByTestId()` to makes tests easier to understand

File: static/src/js/components/GranuleImage/GranuleImage.js [hunk 4]
```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
...
+
+  return (
+    <div className={containerClassName}>
+      { isOpen ? (
+        <Button
+          className="granule-image__button granule-image__button--close"
+          icon={FaTimes}
+          onClick={handleToggleImage}
+          label="Close browse image"
+          data-testid="granule-image__button granule-image__button--close"
```
Comment (macrouch) [inline static/src/js/components/GranuleImage/GranuleImage.js:72]: `granule-image__button` doesn't belong in the testid, think of it like a regular id attribute, its just used for testing

File: static/src/js/components/GranuleImage/GranuleImage.js [hunk 5]
```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
  * @param {Object} props - The props passed into the component.
  * @param {String} imageSrc - The image source.
```
Comment (macrouch) [inline static/src/js/components/GranuleImage/GranuleImage.js:14]: This should be updated, it is no longer how we comment nested parameters. See ABExperiement.js for an example

File: static/src/js/components/GranuleImage/GranuleImage.js [hunk 6]
```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
...
-    if (!imageSrc) return null
-
-    return (
-      <div className={containerClassName}>
+  // When the imageSrc changes value then load the new image
+  useEffect(() => {
+    loadImage()
+  }, [imageSrc])
+
+  if (!imageSrc) return null
```
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply static/src/js/components/GranuleImage/GranuleImage.js:62]: Unfortunately at least as this was yes. Without that return null then the button for the granule image get rendered at the home page even without a focused granule. I can remove this here and put a conditional in the granuleImageContainer if that is preferred?

File: static/src/js/components/GranuleImage/__tests__/GranuleImage.test.js (hunk 2 above)
Comment (eudoroolivares2016) [reply static/src/js/components/GranuleImage/__tests__/GranuleImage.test.js:12]: No I think I was using it earlier and didn't remove it!

File: static/src/js/components/GranuleImage/GranuleImage.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (DanielJDufour) [reply static/src/js/components/GranuleImage/GranuleImage.js:51]: sounds good.  thanks for looking into that. :-)

File: static/src/js/components/GranuleImage/GranuleImage.js (hunk 5 above)
Comment (eudoroolivares2016) [reply static/src/js/components/GranuleImage/GranuleImage.js]: changed

File: static/src/js/components/GranuleImage/GranuleImage.js [hunk 7]
```diff
@@ -13,128 +13,99 @@ import './GranuleImage.scss'
  * @param {Object} props - The props passed into the component.
  * @param {String} imageSrc - The image source.
  */
-class GranuleImage extends Component {
-  constructor(props) {
-    super(props)
```
[the remark this answers is not in the document]
Comment (eudoroolivares2016) [reply static/src/js/components/GranuleImage/GranuleImage.js]: removed

File: static/src/js/components/GranuleImage/GranuleImage.js (hunk 4 above)
Comment (eudoroolivares2016) [reply static/src/js/components/GranuleImage/GranuleImage.js]: removed

**Review by macrouch — APPROVED** [review]

File: static/src/js/components/GranuleImage/GranuleImage.js
Comment (jaybarra) [inline static/src/js/components/GranuleImage/GranuleImage.js:1]: I just noticed these are "js" and not "jsx", do we not use that distinction generally?

Comment (eudoroolivares2016) [reply static/src/js/components/GranuleImage/GranuleImage.js:1]: image
Yea I don't think we have any files as jsx?

**Review by jaybarra — APPROVED** [review]

**Pull request merged**